### PR TITLE
Adjusted the generation of cpp templates to be valid cpp

### PR DESCRIPTION
**Pull request by mamueluth**
- Changed the name for placeholders like $ClassName$ to valid cpp syntax.
- Adjusted workflow to include templates/ dir for cpplint
- Added genuine licenses which is replaced in process. **This is done by deleting the old one in the respective file.** As a result, the  Apache License, Version 2.0 needs to be present in exact form otherwise this could lead to deletion of unintended lines. 
- ~~Currently, `ament_copyright` won't recognize the Apache License, Version 2.0 in the given form. Couldn't find a difference, so I am not exactly sure why this is the case. The c++ files are therefore still excluded from the check. However, `ament_cpplint` checks this anyway (but only in a very rudimentary way).~~ Seems to work now with only one copyright line at beginning
- python files don't include license as of now, but are generated with different scripts. Could be changed in future

### CODE REVIEW

**Comment by mamueluth** [comment]
**NOT** tested so far. Besides, `ament_copyright` won't recognize license...

**Comment by mamueluth** [comment]
Tested, should work. 
However, running the tests for controller and hardware interface  before merging is highly recommended.

File: templates/ros2_control/controller/dummy_controller.hpp [hunk 1]
```diff
@@ -1,14 +1,27 @@
-$LICENSE$
-
-#ifndef $PACKAGE_NAME$__$FILE_NAME$_HPP_
-#define $PACKAGE_NAME$__$FILE_NAME$_HPP_
+// Copyright (c) $YEAR$, $NAME_ON_LICENSE$
+// Copyright (c) $YEAR$, Stogl Robotics Consulting UG (haftungsbeschränkt) (template)
```
Comment (destogl) [inline templates/ros2_control/controller/dummy_controller.hpp:2]:
```suggestion
// Copyright (c) 2022, Stogl Robotics Consulting UG (haftungsbeschränkt)
```

Comment (mamueluth) [reply templates/ros2_control/controller/dummy_controller.hpp]: done

**Comment by mamueluth** [comment]
Please verify that generation of controllers and especially hardware is working as intended.

**Review by destogl — APPROVED** [review]

**Pull request merged**